### ISSUE TITLE
Report an error and solving method

### ISSUE THREAD

**Issue by Hao-Starrr** [UZ-SLAMLab/ORB_SLAM3#566]
This error may occur on Ubuntu 20.04 and 18.04.

When running `./build.sh`, I got 
```
make[2]: *** [CMakeFiles/ORB_SLAM3.dir/build.make:76: CMakeFiles/ORB_SLAM3.dir/src/Tracking.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:711: CMakeFiles/ORB_SLAM3.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```
The problem can be solved by add `add_compile_options(-std=c++14)` before CMakeLists.txt file.
Then it build successfully.

Same solving method for the error:
```
make[2]: *** [CMakeFiles/ORB_SLAM3.dir/build.make:375：CMakeFiles/ORB_SLAM3.dir/src/MLPnPsolver.cpp.o] Error 1
make[2]: *** [CMakeFiles/ORB_SLAM3.dir/build.make:76：CMakeFiles/ORB_SLAM3.dir/src/Tracking.cc.o] Error 1
make[2]: *** [CMakeFiles/ORB_SLAM3.dir/build.make:245：CMakeFiles/ORB_SLAM3.dir/src/Frame.cc.o] Error 1
make[2]: *** [CMakeFiles/ORB_SLAM3.dir/build.make:102：CMakeFiles/ORB_SLAM3.dir/src/LoopClosing.cc.o] Error 1
make[2]: *** [CMakeFiles/ORB_SLAM3.dir/build.make:323：CMakeFiles/ORB_SLAM3.dir/src/G2oTypes.cc.o] Error 1
make[2]: *** [CMakeFiles/ORB_SLAM3.dir/build.make:232：CMakeFiles/ORB_SLAM3.dir/src/Optimizer.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:390：CMakeFiles/ORB_SLAM3.dir/all] Error 2
make: *** [Makefile:84：all] Error 2
```

**Comment by dframirez-usmc** [comment]
Worked for me. 
Solved other issue during `/build.sh` of `g2o` and `Sophus`.
Thank you.

**Comment by anastaga** [comment]
Where exactly should i paste add_compile_options(-std=c++14) in order to make it work?

**Comment by Hao-Starrr** [comment]
> Where exactly should i paste add_compile_options(-std=c++14) in order to make it work?

At the first line of CMakeLists.txt file.

**Comment by CruxDevStuff** [comment]
Worked for me too, Thanks :)

**Comment by MadanGTiwary** [comment]
Worked for me, thanks.

**Comment by kenhuang1964** [comment]
Anyone getting this error?

make[1]: *** [CMakeFiles/Makefile2:148: CMakeFiles/ORB_SLAM3.dir/all] Error 2
make: *** [Makefile:91: all] Error 2

**Comment by jeremy7703** [comment]
> Anyone getting this error?
> 
> make[1]: *** [CMakeFiles/Makefile2:148: CMakeFiles/ORB_SLAM3.dir/all] Error 2 make: *** [Makefile:91: all] Error 2

I have found the same problem with my computer, have you solved this?

**Comment by kenhuang1964** [comment]
> > Anyone getting this error?
> > make[1]: *** [CMakeFiles/Makefile2:148: CMakeFiles/ORB_SLAM3.dir/all] Error 2 make: *** [Makefile:91: all] Error 2
> 
> I have found the same problem with my computer, have you solved this?

Yes, I changed some settings to C++ 14 I think. It's been a while. I forgot lol

**Comment by alanthomas29** [comment]
I had similar issues and updated the cmake file and i could see that after making the changes it builds fine. I will attach a copy my cmakelist.txt file
[CMakeLists.txt](https://github.com/UZ-SLAMLab/ORB_SLAM3/files/15486037/CMakeLists.txt)

**Comment by analwang** [comment]
Sorry, this didn't solve my problem even I have edited CMakeLists.txt as you said,...
but:

make[2]: *** [CMakeFiles/ORB_SLAM3.dir/build.make:76：CMakeFiles/ORB_SLAM3.dir/src/System.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:148：CMakeFiles/ORB_SLAM3.dir/all] Error 2
make: *** [Makefile:91：all] Error 2

My CMakeLists.txt is as follows
[CMakeLists.txt](https://github.com/user-attachments/files/18375334/CMakeLists.txt)

**Comment by jaticona** [comment]
to enforce -std=c++14, you should be add following lines in ORB-SLAM3/CMAkeLists.txt
set(CMAKE_CXX_STANDARD 14)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_EXTENSIONS OFF)
add the lines befor "# Check C++11 or C++0x support"
